libs/sdk-core/src/reverseswap.rs

```diff
@@ -22,68 +33,238 @@ pub struct CreateReverseSwapResponse {
+                    input: txins.clone(),
+                    output: tx_out,
+                };
+
+                let recommended_fees = self.chain_service.recommended_fees().await?;
+                let sat_per_vbyte = recommended_fees.half_hour_fee; // TODO Configurable
+
+                let redeem_script_bytes =
+                    bitcoin::psbt::serialize::Serialize::serialize(&redeem_script);
 
-        // TODO persist
+                // Based on https://github.com/breez/boltz/blob/master/boltz.go#L31
+                let refund_witness_input_size: u32 = 1 + 1 + 8 + 73 + 1 + 32 + 1 + 100;
+                let tx_weight = tx.strippedsize() as u32 * WITNESS_SCALE_FACTOR as u32
+                    + refund_witness_input_size * txins.len() as u32;
+                let fees: u64 = (tx_weight * sat_per_vbyte / WITNESS_SCALE_FACTOR as u32) as u64;
+                tx.output[0].value = confirmed_amount - fees;
+
+                let scpt = Secp256k1::signing_only();
+
+                // Sign inputs (iterate, even though we only have one input)
+                let mut signed_inputs: Vec<TxIn> = Vec::new();
+                for (index, input) in tx.input.iter().enumerate() {
+                    let mut signer = SighashCache::new(&tx);
+                    let sig = signer.segwit_signature_hash(
+                        index,
+                        &redeem_script,
+                        utxos.confirmed[index].value as u64,
+                        EcdsaSighashType::All,
+                    )?;
+                    let msg = Message::from_slice(&sig[..])?;
+                    let secret_key = SecretKey::from_slice(rs.local_private_key.as_slice())?;
+                    let sig = scpt.sign_ecdsa(&msg, &secret_key);
+
+                    let mut sigvec = sig.serialize_der().to_vec();
+                    sigvec.push(EcdsaSighashType::All as u8);
+
+                    let witness: Vec<Vec<u8>> = vec![
+                        sigvec,
+                        rs.local_preimage.clone(),
+                        redeem_script_bytes.clone(),
+                    ];
+
+                    let mut signed_input = input.clone();
+                    let w = Witness::from_vec(witness);
+                    signed_input.witness = w;
+                    signed_inputs.push(signed_input);
+                }
+                tx.input = signed_inputs;
+
+                Ok(tx)
+            }
+            Some(addr_type) => Err(anyhow!("Unexpected lock address type: {addr_type:?}")),
+            None => Err(anyhow!("Could not determine lock address type")),
+        }
+    }
+
+    pub(crate) async fn execute_pending_reverse_swaps(&self) -> Result<()> {
+        let monitored = self.refresh_monitored_reverse_swaps().await?;
+        info!("Found {} monitored reverse swaps", monitored.len());
+
+        // Depending on the new state, decide next steps and transition to the new state
+        for rs in monitored {
```

Comment:
I am missing the case where we started a reverse swap, paid for the old invoice but from some reason the remote swap provider has canceled the payment.
It seems that these swaps will be monitored until expiration and won't marked as canceled.